include/turtle/detail/expectation_template.hpp

```diff
@@ -196,16 +221,41 @@ namespace detail
         {
             return invocation_->verify();
         }
-
+#if defined(MOCK_THREAD_SAFE)
+        bool verify(const boost::shared_ptr< condition_variable> &cv, lock &lk) const
+        {
+            if (timeout_)
+            {
+                while (!invocation_->verify())
```

Comment:
What I meant is this will pass whereas it should have failed

    BOOST_FIXTURE_TEST_CASE( mock_object_asynchonous_call_expectation_in_sequence, mock_error_fixture )
    {
    #if defined(MOCK_ASYNC)
        const mock_class m{};
        MOCK_EXPECT( m.my_tag ).async(MOCK_THREAD_NAMESPACE::chrono::milliseconds(50)).once().with( "some parameter" );
        MOCK_EXPECT( m.my_tag ).async(MOCK_THREAD_NAMESPACE::chrono::milliseconds(5)).once().with( "some parameter2" );
        MOCK_THREAD_NAMESPACE::thread context([&](){
            MOCK_THREAD_NAMESPACE::this_thread::sleep_for(MOCK_THREAD_NAMESPACE::chrono::milliseconds(10));
            m.my_method("some parameter");
            m.my_method("some parameter2");
        });
        mock::verify();
        CHECK_CALLS( 2 );
        context.join();
    #endif
    }

and this will fail whereas it should have succeeded

    BOOST_FIXTURE_TEST_CASE( mock_object_asynchonous_call_expectation_in_sequence, mock_error_fixture )
    {
    #if defined(MOCK_ASYNC)
        const mock_class m{};
        MOCK_EXPECT( m.my_tag ).async(MOCK_THREAD_NAMESPACE::chrono::milliseconds(50)).once().with( "some parameter" );
        MOCK_EXPECT( m.my_tag ).async(MOCK_THREAD_NAMESPACE::chrono::milliseconds(20)).once().with( "some parameter2" );
        MOCK_THREAD_NAMESPACE::thread context([&](){
            MOCK_THREAD_NAMESPACE::this_thread::sleep_for(MOCK_THREAD_NAMESPACE::chrono::milliseconds(10));
            m.my_method("some parameter2");
            MOCK_THREAD_NAMESPACE::this_thread::sleep_for(MOCK_THREAD_NAMESPACE::chrono::milliseconds(20));
            m.my_method("some parameter");
        });
        mock::verify();
        CHECK_CALLS( 2 );
        context.join();
    #endif
    }